airbyte-integrations/connectors/source-stripe/unit_tests/integration/test_persons.py

```diff
@@ -0,0 +1,632 @@
+        assert len(actual_messages.records) == 1
+
+    @HttpMocker()
+    def test_rate_limit_max_attempts_exceeded(self, http_mocker: HttpMocker) -> None:
+        http_mocker.get(
+            _create_request("accounts").with_limit(100).build(),
+            _create_response().with_record(record=_create_record("accounts")).build(),
+        )
+
+        http_mocker.get(
+            _create_request("persons").with_limit(100).build(),
+            [
+                # Used to pass the initial check_availability before starting the sync
+                _create_response().with_record(record=_create_record("persons")).with_record(record=_create_record("persons")).build(),
+                a_response_with_status(429),
+                a_response_with_status(429),
+                a_response_with_status(429),
+                a_response_with_status(429),
+                a_response_with_status(429),
+            ]
+        )
+
+        http_mocker.get(
+            _create_request("events").with_created_gte(_NOW - timedelta(days=30)).with_created_lte(_NOW).with_limit(100).with_types(
+                ["person.created", "person.updated", "person.deleted"]).build(),
+            _create_response().with_record(record=_create_record("events")).with_record(record=_create_record("events")).build(),
+        )
+
+        source = SourceStripe(config=_CONFIG, catalog=_create_catalog())
+        actual_messages = read(source, config=_CONFIG, catalog=_create_catalog())
+
+        assert len(actual_messages.errors) == 1
+
+    @HttpMocker()
+    def test_incremental_rate_limit_max_attempts_exceeded(self, http_mocker: HttpMocker) -> None:
+        state_datetime = _NOW - timedelta(days=5)
+        cursor_datetime = state_datetime + _AVOIDING_INCLUSIVE_BOUNDARIES
+
+        http_mocker.get(
+            _create_request("accounts").with_limit(100).build(),
+            _create_response().with_record(record=_create_record("accounts")).build(),
+        )
+
+        http_mocker.get(
+            _create_request("persons").with_limit(100).build(),
+            _create_response().with_record(record=_create_record("persons")).with_record(record=_create_record("persons")).build(),
+        )
+
+        # Mock when check_availability is run on the persons incremental stream
+        http_mocker.get(
+            _create_request("events").with_created_gte(_NOW - timedelta(days=30)).with_created_lte(_NOW).with_limit(100).with_types(
+                ["person.created", "person.updated", "person.deleted"]).build(),
+            _create_response().with_record(record=_create_record("persons_event_created")).with_record(
+                record=_create_record("persons_event_created")).build(),
+        )
+
+        http_mocker.get(
+            _create_request("events").with_created_gte(cursor_datetime).with_created_lte(_NOW).with_limit(100).with_types(
+                ["person.created", "person.updated", "person.deleted"]).build(),
+            [
```

Comment:
nit: it is fine to show that we return the 429 multiple times but I think the framework will by default always return the last response that was configured so this could have been:

```
        [
                a_response_with_status(429),
        ]
```

... or even just `a_response_with_status(429)`

Let me know if you think this is a behavior that we should prevent

Reply:
okay yeah that makes sense, I'll switch it to that and just comment that this simulates the 5 retries. i think the behavior you mentioned is fine